sway/main.c

```diff
@@ -223,34 +223,35 @@ static void executable_sanity_check() {
 static void drop_permissions(bool keep_caps) {
 	if (getuid() != geteuid() || getgid() != getegid()) {
 		if (setgid(getgid()) != 0) {
-			wlr_log(L_ERROR, "Unable to drop root");
+			sway_log(L_ERROR, "Unable to drop root");
 			exit(EXIT_FAILURE);
 		}
 		if (setuid(getuid()) != 0) {
-			wlr_log(L_ERROR, "Unable to drop root");
+			sway_log(L_ERROR, "Unable to drop root");
 			exit(EXIT_FAILURE);
 		}
 	}
 	if (setuid(0) != -1) {
-		wlr_log(L_ERROR, "Root privileges can be restored.");
+		sway_log(L_ERROR, "Root privileges can be restored.");
 		exit(EXIT_FAILURE);
 	}
 #ifdef __linux__
 	if (keep_caps) {
 		// Drop every cap except CAP_SYS_PTRACE
 		cap_t caps = cap_init();
 		cap_value_t keep = CAP_SYS_PTRACE;
-		wlr_log(L_INFO, "Dropping extra capabilities");
+		sway_log(L_INFO, "Dropping extra capabilities");
 		if (cap_set_flag(caps, CAP_PERMITTED, 1, &keep, CAP_SET) ||
 			cap_set_flag(caps, CAP_EFFECTIVE, 1, &keep, CAP_SET) ||
 			cap_set_proc(caps)) {
-			wlr_log(L_ERROR, "Failed to drop extra capabilities");
+			sway_log(L_ERROR, "Failed to drop extra capabilities");
 			exit(EXIT_FAILURE);
 		}
 	}
 #endif
 }
 
+void wlr_log_init(log_importance_t verbosity, log_callback_t callback);
```

Comment:
Should include the appropriate header instead of doing this

Reply:
that conflicts on the `L_DEBUG` etc. defines =.=

Reply:
Yeah, that's why we need to prefix wlroots symbols.